consensus-types/types/payload.go

```diff
@@ -88,16 +88,17 @@ type ExecutionPayload struct {
 	ExcessBlobGas math.U64 `json:"excessBlobGas"`
 }
 
-func (p *ExecutionPayload) EnsureNotNilWithdrawals() {
-	// Post Shanghai an EL explicitly check that Withdrawals are not nil
-	// (instead empty slices are fine). Currently, BeaconKit duly builds
-	// a block with Withdrawals set to empty slice if there are no
-	// withdrawals) but as soon as the block is returned by CometBFT
-	// for verification, the SSZ decoding sets the empty slice to nil.
-	// This code change solves the issue.
-	if p.Withdrawals == nil {
-		p.Withdrawals = make([]*engineprimitives.Withdrawal, 0)
+func NewEmptyExecutionPayloadWithVersion(forkVersion common.Version) *ExecutionPayload {
+	ep := &ExecutionPayload{
+		Versionable:   NewVersionable(forkVersion),
+		BaseFeePerGas: &math.U256{},
+	}
+
+	// For any fork version after Bellatrix (Capella onwards), non-nil withdrawals are required.
+	if version.IsAfter(forkVersion, version.Bellatrix()) {
+		ep.Withdrawals = make([]*engineprimitives.Withdrawal, 0)
 	}
+	return ep
 }
```

Comment:
This is crude fix I made. It works but I didn't fully get the SSZ related types.
I believe this fits better in the EnsureSyntaxFromSSZ